src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -38,13 +38,13 @@ public bool AllowUnsafeBlocks
         public string ApplicationConfiguration
         {
             set { _store[nameof(ApplicationConfiguration)] = value; }
-            get { return (string)_store[nameof(ApplicationConfiguration)]; }
+            get { return _store[nameof(ApplicationConfiguration)] as string; }
```

Comment:
We should keep the cast vs. as.  As @amcasey pointed out it will identify coding errors that `as` will hide.